### PR TITLE
add freihand dataset

### PR DESCRIPTION
**Pull request by liuxin9608**
Add freihand dataset

### CODE REVIEW

File: tests/test_datasets/test_top_down_dataset.py [hunk 1]
```diff
@@ -224,6 +248,13 @@ def test_top_down_FreiHand_dataset():
     assert custom_dataset.num_images == 8
     _ = custom_dataset[0]
 
+    outputs = load_json_to_output('tests/data/freihand/test_freihand.json')
+    with tempfile.TemporaryDirectory() as tmpdir:
+        infos = custom_dataset.evaluate(outputs, tmpdir, ['PCK', 'EPE', 'AUC'])
+        assert (abs(infos['PCK'] - 1.0) < 1e-4)
+        assert (abs(infos['AUC'] - 0.95) < 1e-4)
```
Comment (innerlee) [inline tests/test_datasets/test_top_down_dataset.py:255]: https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_almost_equal.html

File: mmpose/core/evaluation/top_down_eval.py [hunk 2]
```diff
@@ -23,7 +23,7 @@ def _calc_distances(preds, targets, normalize):
     N, K, _ = preds.shape
     distances = np.full((K, N), -1, dtype=np.float32)
     eps = np.finfo(np.float32).eps
-    mask = (targets[..., 0] > eps) | (targets[..., 1] > eps)
+    mask = (targets[..., 0] > -eps) & (targets[..., 1] > -eps)
```
Comment (innerlee) [inline mmpose/core/evaluation/top_down_eval.py:26]: is this a bug fix?

File: mmpose/core/evaluation/top_down_eval.py [hunk 3]
```diff
@@ -81,8 +81,10 @@ def _get_max_preds(heatmaps):
     preds[:, :, 0] = preds[:, :, 0] % W
     preds[:, :, 1] = preds[:, :, 1] // W
 
-    pred_mask = np.tile(maxvals > 0.0, (1, 1, 2))
-    preds *= pred_mask
+    # pred_mask = np.tile(maxvals > 0.0, (1, 1, 2))
+    # pred_mask = np.tile(np.where(maxvals > 0.0, 1, -1), (1, 1, 2))
+    # preds *= pred_mask
+    preds = np.where(np.tile(maxvals, (1, 1, 2)) > 0.0, preds, -1)
```
Comment (innerlee) [inline mmpose/core/evaluation/top_down_eval.py:87]: is this (return -1 instead of 0) breaking?

Comment (innerlee) [reply mmpose/core/evaluation/top_down_eval.py:87]: okay, the tests have changed accordingly

File: mmpose/core/evaluation/top_down_eval.py [hunk 4]
```diff
@@ -81,8 +81,10 @@ def _get_max_preds(heatmaps):
     preds[:, :, 0] = preds[:, :, 0] % W
     preds[:, :, 1] = preds[:, :, 1] // W
 
-    pred_mask = np.tile(maxvals > 0.0, (1, 1, 2))
-    preds *= pred_mask
+    # pred_mask = np.tile(maxvals > 0.0, (1, 1, 2))
+    # pred_mask = np.tile(np.where(maxvals > 0.0, 1, -1), (1, 1, 2))
```
Comment (innerlee) [inline mmpose/core/evaluation/top_down_eval.py:85]: rm commented lines

**Comment by innerlee** [comment]
looks good

File: mmpose/core/evaluation/top_down_eval.py (hunk 2 above)
Comment (jin-s13) [reply mmpose/core/evaluation/top_down_eval.py:26]: Yes. Keypoints whose coordinates are  (x, 0) or (0, y) should be valid.

**Review by innerlee — APPROVED** [review]

**Pull request merged**